src/Components/Directory.Build.targets

```diff
@@ -30,4 +41,56 @@
                               PackagePath="buildTransitive\$(TargetFramework)\$(PackageId).targets" />
     </ItemGroup>
   </Target>
+
+  <PropertyGroup>
+    <ConfigurationSchemaGeneratorPath>$(ArtifactsBinDir)ConfigurationSchemaGenerator\$(Configuration)\$(NetCurrent)\ConfigurationSchemaGenerator.dll</ConfigurationSchemaGeneratorPath>
+
+    <GeneratedConfigurationSchemaOutputPath>$(IntermediateOutputPath)ConfigurationSchema.json</GeneratedConfigurationSchemaOutputPath>
+  </PropertyGroup>
+  
+  <Target Name="GenerateConfigurationSchema"
+          DependsOnTargets="CoreGenerateConfigurationSchema;
+                            CopyNewConfigurationSchema;
+                            CompareConfigurationSchema" />
+  
+  <Target Name="CoreGenerateConfigurationSchema"
+          Inputs="$(ConfigurationSchemaGeneratorPath);
+                  @(IntermediateAssembly);
+                  @(ReferencePathWithRefAssemblies)"
+          Outputs="$(GeneratedConfigurationSchemaOutputPath)">
+    <PropertyGroup>
+
+      <GeneratorCommandLine>"$(DotNetTool)" exec $(ConfigurationSchemaGeneratorPath)</GeneratorCommandLine>
+      <GeneratorCommandLine>$(GeneratorCommandLine) --input "@(IntermediateAssembly)"</GeneratorCommandLine>
```

Comment:
Haven't looked at it in a while, but TargetsTriggeredByCompilation is guaranteed to be called after CoreCompile, right?

Reply:
Correct. See https://github.com/dotnet/roslyn/blob/5249865bc562b64333ae80de6ccb47e7881b7840/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L84-L173

Whenever CoreCompile runs `Csc`, it will `<CallTarget Targets="$(TargetsTriggeredByCompilation)"`